nexus_constructor/nexus/nexus_wrapper.py

```diff
@@ -204,12 +204,21 @@ def find_depends_on_for_components(_, node):
                 transforms_dependee_of[depends_on_transform] = []
             transforms_dependee_of[depends_on_transform].append(group_name)
 
-        for transform, dependee_of in transforms_dependee_of.items():
-            # numpy should cast to a scalar value if there is just one item.
-            self.nexus_file[transform].attrs["dependee_of"] = np.array(
+        def set_dependee_of_attr(path: str, dependee_of: List[str]):
+            self.nexus_file[path].attrs["dependee_of"] = np.array(
                 dependee_of, dtype=h5py.special_dtype(vlen=str)
             )
 
+        for transform, dependee_of in transforms_dependee_of.items():
+            try:
+                # numpy should cast to a scalar value if there is just one item.
+                # try and use an absolute path
+                set_dependee_of_attr(transform, dependee_of)
+            except KeyError:
+                # try and use a relative path instead
+                path = f"{dependee_of[0]}/{transform.decode('UTF-8')}"
```

Comment:
This can fail in the case that the transformation is a dependee of multiple components. The key of `transforms_dependee_of` _must_ already be the absolute path to the transform, otherwise it is possible to have multiple entries in the dictionary for the same transformation (because the same transformation can have different relative paths). I guess this would result in the dependee_of list getting overwritten and therefore missing entries.

Please add a unit test to cover this situation.
```
ComponentA
    depends_on: Transforms1/transform1
    Transforms1
        transform1
ComponentB
     depends_on: "/entry/instrument/ComponentA/Transforms1/transform1
```